python/ray/tests/test_cli.py

```diff
@@ -979,6 +983,79 @@ def output_ready():
         _check_output_via_pattern("test_ray_status_multinode_v1.txt", result)
 
 
+@pytest.fixture
+def start_open_port_check_server():
+    class OpenPortCheckServer(BaseHTTPRequestHandler):
+        request_ports = None
+        response_open_ports = []
+
+        def do_POST(self):
+            content_length = int(self.headers["Content-Length"])
+            post_data = self.rfile.read(content_length)
+            payload = json.loads(post_data)
+            OpenPortCheckServer.request_ports = payload["ports"]
+            self.send_response(200)
+            self.send_header("Content-Type", "application/json")
+            self.end_headers()
+            self.wfile.write(
+                json.dumps(
+                    {
+                        "open_ports": OpenPortCheckServer.response_open_ports,
+                        "checked_ports": payload["ports"],
+                    }
+                ).encode("utf-8")
+            )
+
+    server = HTTPServer(("127.0.0.1", 9000), OpenPortCheckServer)
+    server_thread = threading.Thread(target=server.serve_forever)
+    server_thread.start()
+
+    yield OpenPortCheckServer
```

Comment:
```suggestion
    server = HTTPServer(("127.0.0.1", 0), OpenPortCheckServer)
    server_thread = threading.Thread(target=server.serve_forever)
    server_thread.start()

    yield OpenPortCheckServer, f"http://{server.server_address[0]}:{server.server_address[1]}"
```

I'd probably do something like this that way we don't have port conflicts if there's two of these running (e.g. if we parallelize this test)